universe/worlds/worlds.go

```diff
@@ -77,6 +79,79 @@ func (w *Worlds) GetWorlds() map[uuid.UUID]universe.World {
 	return worlds
 }
 
+func (w *Worlds) GetOptions(spaces map[uuid.UUID]universe.Space) ([]dto.ExploreOption, error) {
+	options := make([]dto.ExploreOption, 0, len(spaces))
+
+	for _, space := range spaces {
+		var description any
+
+		nameAttributeID := entry.NewAttributeID(universe.GetSystemPluginID(), universe.SpaceAttributeNameName)
+		nameValue, nameOk := space.GetSpaceAttributeValue(nameAttributeID)
+		if !nameOk {
+			return nil, errors.Errorf("could not get name value %q", nameValue)
+		}
+
+		if nameValue == nil {
+			return nil, errors.Errorf("spaceValue not found %q", nameValue)
+		}
+
+		descriptionAttributeID := entry.NewAttributeID(universe.GetSystemPluginID(), universe.SpaceAttributeNameDescription)
+		descriptionValue, _ := space.GetSpaceAttributeValue(descriptionAttributeID)
+
+		name := (*nameValue)[universe.SpaceAttributeNameName]
+
+		if descriptionValue != nil {
+			description = (*descriptionValue)[universe.SpaceAttributeNameDescription]
+		} else {
+			description = nil
+		}
+
+		subSpaces := space.GetSpaces(false)
+		subOptions, err := w.GetSubOptions(subSpaces)
```

Comment:
ok, you made it for deep level 2, what about 3? will write another one method for it? and for level 4 too?